tests/system/tokenizer/test_cohere_api_tokenizer.py

```diff
@@ -1,91 +1,56 @@
 
     @staticmethod
     @pytest.fixture
     def text():
-        return "Hello World!"
+        return "string with special characters like !@#$%^&*()_+日本 " \
+               "spaces   \n \n\n CASE cAse "
+
 
     @staticmethod
     @pytest.fixture
     def expected_tokens(text):
-        return ["Hello", " World", "!"]
-
-    # region: test tokenize
-
-    @staticmethod
-    def test_tokenize(tokenizer, text, expected_tokens):
-        tokens = tokenizer.tokenize(text)
-        assert tokens == expected_tokens, f"\nExpected: {expected_tokens}" \
-                                          f"\nActual: {tokens}"
-
-    @staticmethod
-    def test_tokenize_empty_string(tokenizer):
-        assert tokenizer.tokenize("") == []
+        return ['string', ' with', ' special', ' characters', ' like',
+                ' !', '@', '#', '$', '%', '^', '&', '*', '()', '_', '+', '日',
+                '本', ' spaces', '   ', '\n ', '\n\n', ' CASE', ' c', 'A',
+                'se', " "]
 
     @staticmethod
-    def test_tokenize_invalid_input_type_raise_exception(tokenizer):
-        with pytest.raises(Exception):
-            tokenizer.tokenize(1)
-
-        with pytest.raises(Exception):
-            tokenizer.tokenize(["asd"])
-
-    # endregion
+    def test_messages_token_count(tokenizer):
+        messages = [MessageBase(role=Role.USER, content="Hello, assistant.")]
+        assert tokenizer.messages_token_count(messages) == 11
 
-    # region: test detokenize
-
-    @staticmethod
-    def test_detokenize(tokenizer, text, expected_tokens):
-        text = tokenizer.detokenize(expected_tokens)
-        assert text == text
-
-    @staticmethod
-    def test_detokenize_empty_string(tokenizer):
-        assert tokenizer.detokenize([]) == ""
-
-    @staticmethod
-    def test_detokenize_invalid_input_type_raise_exception(tokenizer):
-        with pytest.raises(Exception):
-            tokenizer.detokenize(1)
-
-        with pytest.raises(Exception):
-            tokenizer.detokenize("asd")
-
-    # region test token_count
-
-    @staticmethod
-    def test_token_count(tokenizer, text, expected_tokens):
-        token_count = tokenizer.token_count(text)
-        assert token_count == len(expected_tokens)
-        assert token_count == len(tokenizer.tokenize(text))
-
-    @staticmethod
-    def test_token_count_empty_string(tokenizer):
-        assert tokenizer.token_count("") == 0
+        messages = [MessageBase(role=Role.USER,
+                                content="Hello, assistant."),
+                    MessageBase(role=Role.ASSISTANT,
+                                content="Hello, user. How can I assist you?")]
+        assert tokenizer.messages_token_count(messages) == 25
 
     @staticmethod
     def test_messages_token_count_empty_messages(tokenizer):
         assert tokenizer.messages_token_count([]) == 3
 
-    # endregion
-
-    # region special tokens
-
     @staticmethod
     def test_special_tokens_to_natural_text(tokenizer):
```

Comment:
@scottmx81 you'd notice that this test case fails, for both the API tokenizer and the HF tokenizer.  
The goal of this test is to verify that if the user's input text happens to include the string "<BOS_TOKEN>" (or any other special string), it is treated like an actual string, not a special token. We added this test after we've been trying to index in Canopy some documents that happened to be about LLMs and tokenizers, so they naturally included those strings. 

I would appreciate if you could check with Cohere people if there's a way to configure the tokenizer to ignore strings like `"<BOS_TOKEN>"` when they are part of the user's input, treating them as regular string syntax.  
Otherwise - we would have to `skip()` this test case until we have a better fix